test/asn1_encode_test.c

```diff
@@ -381,7 +381,7 @@ static ASN1_INT64_DATA int64_expected[] = {
     CUSTOM_EXPECTED_SUCCESS(0x80000000, 0x80000000), /* t_4bytes_1 */
     CUSTOM_EXPECTED_SUCCESS(INT32_MAX - 1, INT32_MAX -1), /* t_4bytes_2 */
     CUSTOM_EXPECTED_FAILURE,     /* t_4bytes_3_pad (illegal padding) */
-    CUSTOM_EXPECTED_SUCCESS(0x80000000, 0x80000000), /* t_4bytes_4_neg */
+    CUSTOM_EXPECTED_SUCCESS(INT32_MIN, INT32_MIN), /* t_4bytes_4_neg */
```

Comment:
Let's talk even about t_8bytes_1. It's no good for int64, too large,  but fine for uint64, but check for INT64_MAX is kind of counterintuitive. I mean it ought to be implicit cast to uint64_t (right?), so maybe explicit cast in initiallizer would be more appropriate?